intellij/org.eclipse.xtext.idea/src/org/eclipse/xtext/idea/filesystem/IdeaWorkspaceConfigProvider.xtend

```diff
@@ -46,50 +56,56 @@ class IdeaWorkspaceConfig implements IWorkspaceConfig {
 
 	val Project project
 
-	override findProjectByName(String name) {
+	override IdeaModuleConfig findProjectByName(String name) {
 		val module = ApplicationManager.application.<Module>runReadAction[
-			ModuleManager.getInstance(project).findModuleByName(name)
+			ModuleManager.getInstance(project).findModuleByName(name)	
 		]
-		if (module == null)
-			return null
-		if (!module.isUsable) 
-			return null
-		return new IdeaModuleConfig(module)
+		if (module == null) return null
+
+		val contentRoot = ModuleRootManager.getInstance(module).contentRoots.head
+		if (contentRoot == null) return null
+
+		return new IdeaModuleConfig(module, contentRoot)
 	}
 
-	override findProjectContaining(URI member) {
+	override IdeaModuleConfig findProjectContaining(URI member) {
 		val file = VirtualFileManager.instance.findFileByUrl(member.toString)
-		if (file == null)
-			return null
-		val module = ProjectRootManager.getInstance(project).fileIndex.getModuleForFile(file)
-		if (module == null)
-			return null
-		if (!module.isUsable) 
-			return null
-		return new IdeaModuleConfig(module)
+		if (file == null) return null
+			
+		val fileIndex = ProjectRootManager.getInstance(project).fileIndex
+		val module = fileIndex.getModuleForFile(file)
+		if (module == null) return null
+		
+		val contentRoot = fileIndex.getContentRootForFile(file)
+		if (contentRoot == null) return null
+			
+		return new IdeaModuleConfig(module, contentRoot)
 	}
-
-
-	private def isUsable(Module module) {
-		!ModuleRootManager.getInstance(module).contentEntries.isEmpty
+	
+	override getPath() {
+		URI.createURI(project.baseDir.url).toFolderURI
```

Comment:
I don't understand why this is a meaningful information in idea? The project file can be put anywhere and is therefore unrelated to every known virtual file, isn't it?